### PR TITLE
Avoid iterating over hashmaps in astconv

### PR DESCRIPTION
**Pull request by Noratrieb**

### CODE REVIEW

**Review by cjgillot — COMMENTED** [review]
Is this enough to remove the allow(rustc::potential_query_instability) in this crate ?

**Comment by Nilstrieb** [comment]
> warning: `rustc_hir_analysis` (lib) generated 20 warnings

no, they're still all over the place.

File: compiler/rustc_hir_analysis/src/astconv/object_safety.rs [hunk 1]
```diff
@@ -148,8 +148,7 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             }
         }
 
-        // Use a `BTreeSet` to keep output in a more consistent order.
```
Comment (petrochenkov) [inline compiler/rustc_hir_analysis/src/astconv/object_safety.rs:151]: This comment still looks relevant.

**Comment by petrochenkov** [comment]
r=me with restoring the comment (https://github.com/rust-lang/rust/pull/117828#discussion_r1397847086), or maybe using `FxIndexSet` there because the sorting doesn't seem necessary there, just the order stability.
@rustbot author

**Pull request merged**